frontend/src/pages/phone.page.tsx

```diff
@@ -30,13 +31,32 @@ const Phone: NextPage = () => {
   const relayNumberData = useRelayNumber();
   const [isInOnboarding, setIsInOnboarding] = useState<boolean>();
   const realPhoneData = useRealPhonesData();
+  // The user hasn't completed the onboarding yet if...
+  const isNotSetup =
+    // ...they haven't purchased the phone plan yet, or...
+    profile?.has_phone === false ||
+    // ...the API request for their Relay number has completed but returned no
+    // result, or...
+    (!relayNumberData.isValidating &&
+      typeof relayNumberData.error === "undefined" &&
+      typeof relayNumberData.data === "undefined") ||
+    // ...there was a list of Relay numbers, but it was empty:
+    relayNumberData.data?.length === 0;
 
   useEffect(() => {
-    // check if phone flag is active - return to premium page if not.
-    if (!isFlagActive(runtimeData.data, "phones")) {
+    if (!runtimeData.data) {
+      return;
+    }
+    if (
+      // Send the user to /premium if the phones flag is disabled...
+      !isFlagActive(runtimeData.data, "phones") ||
+      // ...or if a phone subscription is not available in the current country,
+      // and the user has not set it up before (possibly in another country):
+      (!isPhonesAvailableInCountry(runtimeData.data) && isNotSetup)
```

Comment:
Out of curiousity, is `!isPhoneAvailabeInCountry` not a sufficient enough condition to redirect to `/premium`?

Reply:
It could be, but I was thinking that it might be annoying if someone has purchased phones, then goes abroad and then all of a sudden isn't able to manage their phones. Though we'll still have that problem in the sense that the menu items will be gone and all that.

Reply:
Good call out